moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp

```diff
@@ -176,6 +180,10 @@ void TrajectoryExecutionManager::initialize()
 
   reconfigure_impl_ = new DynamicReconfigureImpl(this);
 
+  ros::NodeHandle private_node_handle("~/trajectory_execution");
+  private_node_handle.getParam("joints_allowed_start_tolerance", joints_allowed_start_tolerance_);
+  updateJointsAllowedStartToleranceEmpty();
```

Comment:
As we can't modify those parameters via `dynamic reconfigure`, I suggest loading those values again, each time _before_ a trajectory shall be executed. Actually, you could/should drop the C++ API to set those parameters then, because they would get overwritten.